runtime/caml/stack.h

```diff
@@ -49,30 +56,32 @@
 #define Wosize_gc_regs (13 /* int regs */ + 16 /* float regs */)
 #define Saved_return_address(sp) *((intnat *)((sp) - 8))
 #ifdef WITH_FRAME_POINTERS
-#define Pop_frame_pointer(sp) (sp) += sizeof(value)
+#define First_frame(sp) ((sp) + 16)
 #else
-#define Pop_frame_pointer(sp)
+#define First_frame(sp) ((sp) + 8)
```

Comment:
I'm a bit perplexed, since this macro would work for any frame, and returns the beginning of the frame by jumping over the return address and possibly stored frame pointer, should it be called something like Frame_begin?

Reply:
On POWER this macro does NOT work for any frame, since the return address is not stored where you expect it.  See https://github.com/xavierleroy/ocaml/blob/resurrect-power/asmcomp/power/NOTES.md for details.

The interface I propose in this PR gives the minimal operations needed to traverse OCaml stacks, for GC or for backtraces.  The `First_frame` macro is to be used when starting the traversal and nowhere else -- even if it happens to do something sensible on non-POWER platforms.  Its name reflects that.

Reply:
Makes sense. Thanks for the explanation!